src/shared/ptyfwd.c

```diff
@@ -515,19 +516,35 @@ static int pty_forward_ansi_process(PTYForward *f, size_t offset) {
                                  * codepoint, and that would create ambiguity. Various terminal emulators
                                  * similar do not support it. */
 
-                                if (IN_SET(c, '\x07', '\x1b')) {
-                                        r = insert_window_title_fix(f, i+1);
-                                        if (r < 0)
-                                                return r;
-
-                                        i += r;
-                                }
+                                r = insert_window_title_fix(f, i+1);
+                                if (r < 0)
+                                        return r;
+                                i += r;
 
+                                f->osc_sequence = mfree(f->osc_sequence);
+                                f->ansi_color_state = ANSI_COLOR_STATE_TEXT;
+                        } else if (c == '\x1b')
+                                /* See the comment above. */
```

Comment:
the comment above need updating now. this part needs to be removed:

```
                                 * since we cannot look ahead to see if the Esc is followed by a "\"
                                 * we cut a corner here and assume it will be "\"e.
```